src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -8,7 +8,28 @@
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws CustomeException {
+        if (user.getAge() == null || user.getPassword() == null || user.getLogin() == null) {
+            throw new CustomeException("User login,password or age is null!");
+        }
+        if (user.getLogin().isEmpty() || (user.getPassword().isEmpty())) {
+            throw new CustomeException("User login or password is empty!");
+        }
+        if ((user.getLogin().length()) < 6) {
+            throw new CustomeException("User login is need to be at least 6 letters! - "
+                    + user.getLogin());
+        }
+        if (user.getAge() < 18) {
+            throw new CustomeException("User age is under 18! " + user.getAge());
+        }
+        if ((user.getPassword().length() < 6)) {
+            throw new CustomeException("User password is need to be at least 6 letters! - "
```

Comment:
The message 'User password is need to be at least 6 letters!' is grammatically incorrect. Consider rephrasing it to 'User password must be at least 6 characters long.'